DOCS-2315 Getting Started with Multistep API Tests Edits (#11307)
DataDog/documentation: content/en/getting_started/synthetics/browser_test.md

    - link: '/synthetics/browser_tests'
      tag: 'Documentation'
      text: 'Learn more about browser tests'
    - link: '/synthetics/identify_synthetics_bots'
      tag: 'Documentation'
      text: 'Learn how to identify Synthetic bots for API tests'
      
---

## Overview

[Browser tests][1] are scenarios that Datadog executes on your web applications. You can configure periodic intervals to run tests from multiple locations, devices, and browsers as well as execute them from your CI/CD pipelines. These tests verify that your users can perform **key business transactions** on your applications and that they are not negatively impacted by the most recent code changes.

## Create a browser test

The example below demonstrates the creation of a browser test that maps a user's journey from adding an item to a cart to successfully checking out. 

{{< img src="getting_started/synthetics/browser-test.png" alt="Browser test" style="width:100%;" >}}

### Configure your test details

> 1. In the Datadog application, hover over **[UX Monitoring][2]** in the left hand menu and select **[Synthetic Tests][2]**.

Author: Our Vale Linter suggests "site" instead of "application".

```suggestion
1. In the Datadog site, hover over **[UX Monitoring][2]** in the left hand menu and select **[Synthetic Tests][2]**.
```

2. In the top right corner, click the **New Test** button.
3. Select **Browser Test**.
4. Define your browser test:

    - Add the URL of the website you want to monitor. If you don’t know what to start with, you can use `https://www.shopist.io`, a test e-commerce web application.
    - Select **Advanced Options** to use custom request headers, authentication credentials, or cookies. 
    - Name your test and set tags to it such as `env:prod` and `app:shopist`. Tags allow you to keep your test suite organized and quickly find tests you're interested in on the homepage.
    - Choose the browsers and devices you want to test with. 

#### Select locations

Select one or more **Managed Locations** or **[Private Locations][3]** to run your test from.

Managed locations allow you to test public-facing websites and endpoints. To test internal applications or simulate user behavior in discrete geographic regions, select one of your private locations instead.

> The Shopist application is publicly available, you can consequently go ahead and pick any managed locations you want to execute your test from.

Author: ```suggestion
The Shopist application is publicly available at `https://www.shopist.io/` and you can pick any managed locations to execute your test from.
```

#### Specify test frequency

> Select the frequency at which you want your test to execute. You can leave the default frequency of 1 hour.

Author: ```suggestion
Select the frequency at which you want your test to execute. You can leave the default frequency of one hour.
```

#### Notify your team

> Design your alert message and add any email address you want your test to send alerts to.

Author: ```suggestion
Design your alert message and add an email address you want your test to send alerts to.
```

{{< img src="getting_started/synthetics/configured-browser-test.mp4" alt="Example browser test configuration" video="true"  >}}

> You can also use [notifications integrations][5] such as Slack, PagerDuty, Microsoft Teams, and webhooks. Note that in order to be able to trigger a Synthetic alert to these notification tools, you first need to set up the corresponding [integration][12].

Author: ```suggestion
You can also use [notifications integrations][5] such as Slack, PagerDuty, Microsoft Teams, and webhooks. In order to trigger a Synthetic alert to these notification tools, you first need to set up the corresponding [integration][12].
```

**Note**: The example website regularly throws an error causing it to intentionally fail. If you include your email address in the **Notify your team** field, you receive an email notification when the test fails and recovers.

## Look at test results

> The Browser Test details page displays an overview of your test configuration, the global and per location uptime, graphs about time-to-interactive and test duration, sample successful and failed test results, and the list of all test results. Depending on the length of your test, you might have to wait for a few minutes to see the first test results come in.

Author: ```suggestion
The **Browser Test** details page displays an overview of your test configuration, the global and per location uptime, graphs about time-to-interactive and test duration, sample successful and failed test results, and the list of all test results. Depending on the length of your test, you might have to wait for a few minutes to see the first test results come in.
```
